src/service.js

```diff
@@ -597,6 +601,34 @@ export function adoptServiceForEmbedDoc(ampdoc, id) {
   );
 }
 
+/**
+ * This adopts the service **factory** from the parent.
+ *
+ * This function is safer than sharing the service instance, since each ampdoc
+ * will create its own instance of the factory (and each instance will have its
+ * own instance data). Note that static data is still shared, so it's not 100%
+ * foolproof.
+ *
+ * @param {!./service/ampdoc-impl.AmpDoc} ampdoc
+ * @param {string} id
+ */
+export function adoptServiceFactoryForEmbedDoc(ampdoc, id) {
+  const parentHolder = getAmpdocServiceHolder(devAssert(ampdoc.getParent()));
+  devAssert(
+    isServiceRegistered(parentHolder, id),
+    `Expected service ${id} to be registered`
+  );
+  const service = getServices(parentHolder)[id];
+  registerServiceInternal(
+    getAmpdocServiceHolder(ampdoc),
+    ampdoc,
+    id,
+    devAssert(service.ctor),
+    /* override */ false,
+    /* adopted */ true
```

Comment:
It actually shouldn't be "adopted". The "adopted" only controls which services are disposed. In case when the instance is shared - we can't dispose the service for FIE - it will dispose it for everyone. So it should be `false` here. If would make it clearer, we could renamed "adopted" to "shared". This would be good to specifically test for.

Reply:
Done, working on the tests now.